### DISCUSSION TITLE
[vps 部署] 修改blog.config.js无效

### DISCUSSION THREAD

**Discussion by 668168** [tangly1024/NotionNext#1963] [Q&A]
**描述bug**
简单说明bug的现象、相关的错误提示、日志等
目前我确定的是如下 4 个变量在 vps 自助部署情况下, 在 blog.config.js 修改无效. 
```
NEXT_PUBLIC_THEME='next'
NEXT_PUBLIC_FIREWORKS=true
NEXT_PUBLIC_SAKURA=true
NEXT_PUBLIC_WIDGET_PET=false
```

**复现步骤**
 1. 在vercel.app 部署, 修改blog.config.js , 这4 个变量生效**正常**.
 2. 在 vps  git clone NotionNext, 修改blog.config.js , 这4 个变量**无法生效**.
 3. 在 vps  git clone NotionNext, 修改.env.local , 这4 个变量生效**正常**.

ps. 所有测试都是重建匿名浏览器测试的

**期望的正常结果**
避免其他人遇到这个 bug, 建议修改

**环境**

- 操作系统: [例如. iOS, Android, macOS, windows]
- 浏览器 [例如. chrome, safari, firefox]
- NotionNext版本 [e.g. 3.13.6]

MacOS + VPS-ubuntu20.04
Chrome 无痕模式
NotionNext version 3.14.0

**Comment by tangly1024** [comment]
修改完 blog.config.js 需要重新 yarn build 才能生效。

推荐修改.env.local ，环境变量修改不需要重新编译

**Comment by 668168** [comment]
"修改完 blog.config.js 需要重新 yarn build 才能生效。"  ==> `yarn build` 后依旧不生效.
要不我启个临时 vps ,您上来复现?

**Comment by tangly1024** [comment]
`在 vps git clone NotionNext, 修改blog.config.js , 这4 个变量无法生效.`

为避免误解，我想确认一下复现这个bug操作的顺序是否如下：

1. git clone
2. vim blog.config.js 
3. 分别修改
https://github.com/tangly1024/NotionNext/blob/e5d71910ed15a216e9fb75bafa8b993722ed1f5b/blog.config.js#L8
https://github.com/tangly1024/NotionNext/blob/e5d71910ed15a216e9fb75bafa8b993722ed1f5b/blog.config.js#L129
https://github.com/tangly1024/NotionNext/blob/e5d71910ed15a216e9fb75bafa8b993722ed1f5b/blog.config.js#L139
https://github.com/tangly1024/NotionNext/blob/e5d71910ed15a216e9fb75bafa8b993722ed1f5b/blog.config.js#L153
使其值分为
```
  THEME: process.env.NEXT_PUBLIC_THEME || 'next', 
  FIREWORKS: process.env.NEXT_PUBLIC_FIREWORKS || false,
  SAKURA: process.env.NEXT_PUBLIC_SAKURA || true,
  WIDGET_PET: process.env.NEXT_PUBLIC_WIDGET_PET || false,
```
4. 执行 `yarn build`，按照当前配置编译打包
5. 确认系统无设置其它环境变量、例如：NEXT_PUBLIC_THEME、NEXT_PUBLIC_FIREWORKS、NEXT_PUBLIC_SAKURA、NEXT_PUBLIC_WIDGET_PET
6. 最后 `yarn start`

**Comment by tangly1024** [comment]
另外，非常感谢您认真的issue❤️

**Comment by 668168** [comment]
谢谢, 的确是这样的配置, 然后在 vps 上部署,复现问题.

最简单的判断方法:
比如桌面的宠物插件, 虽然关闭了, 但还是有.

**Comment by dishengai** [comment]
> 修改完 blog.config.js 需要重新 yarn build 才能生效。
> 
> 推荐修改.env.local ，环境变量修改不需要重新编译

请问一下 如果不会以上,直接在网页端的github可以修改完成吗?

**Comment by dz95283** [comment]
[](url)
